src/lib/meilisearch.ts

```diff
@@ -253,14 +256,68 @@ class MeiliSearch {
   ///
 
   /**
-   * Get private and public key
+   * Get all API keys
    * @memberof MeiliSearch
    * @method getKey
-   * @returns {Promise<Keys>} Promise containing an object with keys
+   * @returns {Promise<Keys>} Promise returning an object with keys
    */
-  async getKeys(): Promise<Keys> {
+  // FIXME: should be Result<Key[]>>
+  async getKeys(): Promise<Key[]> {
     const url = `keys`
-    return await this.httpRequest.get<Keys>(url)
+    return await this.httpRequest.get<Key[]>(url)
+  }
+
+  /**
+   * Get one API key
+   * @memberof MeiliSearch
+   * @method getKey
+   *
+   * @param {string} key - Key
+   * @returns {Promise<Keys>} Promise returning a key
+   */
+  async getKey(key: string): Promise<Key> {
+    const url = `keys/${key}`
+    return await this.httpRequest.get<Key>(url)
+  }
+
+  /**
+   * Create one API key
+   * @memberof MeiliSearch
+   * @method createKey
+   *
+   * @param {KeyPayload} options - Key options
+   * @returns {Promise<Key>} Promise returning an object with keys
+   */
+  async createKey(options: KeyPayload): Promise<Key> {
+    const url = `keys`
+    return await this.httpRequest.post(url, options)
+  }
+
+  /**
+   * Update one API key
+   * @memberof MeiliSearch
+   * @method updateKey
+   *
+   * @param {string} key - Key
+   * @param {KeyPayload} options - Key options
+   * @returns {Promise<Key>} Promise returning an object with keys
+   */
+  async updateKey(key: string, options: KeyPayload): Promise<Key> {
+    const url = `keys/${key}`
+    return await this.httpRequest.patch(url, options)
+  }
+
+  /**
+   * Delete one API key
+   * @memberof MeiliSearch
+   * @method getKey
+   *
+   * @param {string} key - Key
+   * @returns {Promise<Key>} Promise returning a key
```

Comment:
```suggestion
   * @returns {Promise<Void>}
```
It could return a response but not a Key